extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+        updateProperties(deviceProperties);
+    }
+
+    private void getTemprature() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/temperature");
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponseTemperature = coapClient.get(TEXT_PLAIN);
+
+        String temperatureText = clientResponseTemperature.getResponseText();
+        String temperatureFirstHalf = temperatureText.substring(0, temperatureText.lastIndexOf(','));
+        String temperature = temperatureFirstHalf.substring(temperatureFirstHalf.lastIndexOf(',') + 1);
+
+        updateState(stringChannelTemperatureUID, new DecimalType(temperature));
+
+        logger.debug(clientResponseTemperature.advanced().getSource().toString());
+    }
+
+    private void getHumidity() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/humidity");
+
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponseHumidity = coapClient.get(TEXT_PLAIN);
+        String humidityText = clientResponseHumidity.getResponseText();
+        String humidityFirstHalf = humidityText.substring(0, humidityText.lastIndexOf(','));
+        String humidity = humidityFirstHalf.substring(humidityFirstHalf.lastIndexOf(',') + 1);
+
+        updateState(stringChannelHumidityUID, new DecimalType(humidity));
+
+        logger.debug(clientResponseHumidity.advanced().getSource().toString());
+    }
+
+    private void getPressure() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/pressure");
+
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponsePressure = coapClient.get(TEXT_PLAIN);
+        String pressureText = clientResponsePressure.getResponseText();
+        String pressureFirstHalf = pressureText.substring(0, pressureText.lastIndexOf(','));
+        String pressure = pressureFirstHalf.substring(pressureFirstHalf.lastIndexOf(',') + 1);
+
+        updateState(stringChannelPressureUID, new DecimalType(pressure));
+
+        logger.debug(clientResponsePressure.advanced().getSource().toString());
+    }
+
+    private void putLCD(String message) {
+        coapClient.setURI("coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/COAP/LCDResource");
+        CoapResponse clientResponseLCD = coapClient.put(message, TEXT_PLAIN);
+
+        logger.debug(clientResponseLCD.advanced().getSource().toString());
+    }
+
+    private void putLightOnOff(String message) {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/actuators/actuator1");
+        StringBuilder coapLightSwitch = new StringBuilder();
+        if (message.compareTo("ON") == 0) {
+            coapLightSwitch.append(",1,pp,100,");
```

Comment:
Since there is nothing more which gets appended to the StringBuilder this also might be a simple String.